### DISCUSSION TITLE
Connecting UniMath to Zenodo

### DISCUSSION THREAD

**Discussion by nmvdw** [UniMath/UniMath#1684] [General]
[Zenodo](https://zenodo.org) is a place where one can upload scientific datasets/software for permanent storage with a persistent digital identifier. Every Zenodo upload is given an DOI and permanently stored. As an example, Coq is on Zenodo: https://zenodo.org/record/7313584#.ZD074y8RphA. Here one can also obtain a bibtex entry for Coq.

Github repositories can be connected to Zenodo (https://docs.github.com/en/repositories/archiving-a-github-repository/referencing-and-citing-content, https://help.zenodo.org). This allows one to create an archive of the repository on Zenodo, and that archive would then get an DOI. We could let every release have an DOI (as in https://github.com/UniMath/UniMath/issues/1631). Github actions is able to do a large part of the work (I heard).

I think that it would be a good idea to connect UniMath to Zenodo. We regularly publish papers that make use of UniMath, and with Zenodo, these papers can cite to a permanent stable copy of UniMath. This would guarantee that the code in the papers we write now, can still be found in the future. Note that the development would still happen in Github, and the Github repository would stay almost identical (the difference is that a badge would be added with a DOI and a link to Zenodo). The purpose of using Zenodo is purely for citations.

What do we all think of this?

**Comment by benediktahrens** [comment]
Sounds great!

**Comment by nmvdw** [comment]
To connect UniMath to Zenodo, a new release must be created. Do we agree to the following release: https://github.com/UniMath/UniMath/releases/tag/untagged-8d612e1d64a08f223bb0?

**Reply by benediktahrens** [reply]
I think the date format is not consistent with previous releases: it should be 2023-04-20 instead of 20-04-2023.

Does one get a separate DOI for each release?

**Reply by nmvdw** [reply]
Yes, every release gets a separate DOI.

**Reply by nmvdw** [reply]
The date is fixed now

**Reply by benediktahrens** [reply]
Awesome! Could you also make sure the tag (which I cannot see in the draft) has the same format as previous tags? This could help when listing tags, to order them.

**Reply by rmatthes** [reply]
The release link is now gone. The moment seems well chosen for this release - no PR appears to be imminent.

**Reply by nmvdw** [reply]
The fixed release link: https://github.com/UniMath/UniMath/releases/edit/untagged-a122a1c135cc4d3cc2de

I had the following tag: v20230420
<img width="938" alt="Screenshot 2023-04-20 at 11 37 55" src="https://user-images.githubusercontent.com/24875025/233325532-94f472d6-9c34-4633-857e-384c80979ef7.png">

**Reply by benediktahrens** [reply]
Looks great to me. Thanks for setting this up.

**Reply by rmatthes** [reply]
LGTM (G stands for great)

**Reply by nmvdw** [reply]
It is here now: https://zenodo.org/record/7848572#.ZEEKWy8RphA

I will add the badge to the repository. The DOI still needs to be archived though, so that one will be added later.

**Comment by rmatthes** [comment]
The Zenodo entry is https://zenodo.org/record/7848572 - is there a favorite form (for Coq, there were cryptic decorations after the main identifier, and in the link posted by Niels seconds before my post as well)?
